backend/push/src/config/amqp-connection.ts

```diff
@@ -0,0 +1,47 @@
+import { CONFIG } from './config';
+import { connect, Channel, Connection } from 'amqplib';
+import { AmqpQueue } from '~/shared/enums/enums';
+import { logger } from './logger';
+import { timeout } from '../helpers/timeout';
+
+const tryConnect = async (tries = 0): Promise<Connection> => {
+  const amqpServer = CONFIG.rabbitmqUrl;
+  if (tries >= 5) {
+    logger.error('Connection failed');
+    process.exit(0);
+  }
+  try {
+    const connection = await connect(amqpServer);
+    connection.on('close', async () => {
+      logger.warn('Rabbitmq connection lost');
+      await timeout(3000);
+      return tryConnect(++tries);
+    });
+    logger.info('Connect to RabbitMQ success!');
+    return connection;
+  } catch {
+    logger.info('Trying to reconnect...');
+    await timeout(3000);
+    return tryConnect(++tries);
```

Comment:
Get rid of duplication

Reply:
i dont get it how i can do that, move this code into function?

Reply:
Yes